### DISCUSSION TITLE
vcpkg_install_msbuild and project that uses vcpkg in Manifest mode

### DISCUSSION THREAD

**Discussion by vividos** [microsoft/vcpkg#29858] [Q&A] [answered]
Hey there,
I'm currently trying to create my own vcpkg registry with a port of a project of mine. The project is a static library that is built using MSBuild and itself uses vcpkg in Manifest mode. My initial port looks like this:
https://github.com/vividos/vcpkg-registry/tree/main/ports/vividos-ulibcpp
The port file downloads the latest release and tries to build it with `vcpkg_install_msbuild`. As I thought building the project using Visual Studio downloads the vcpkg referenced projects (some Boost libraries), `vcpkg_install_msbuild` would do the same. Unfortunately, the project doesn't create a `vcpkg_installed` folder and doesn't install Boost. Compiling then fails since it can't find some Boost header (boost/date_time.hpp).
Do I have to pass some more MSBuild properties for this to work? Or some other changes to the port? Maybe specify the dependencies?
Thanks!
Michael

**Comment by Osyotr** [comment]
> Maybe specify the dependencies?

This.

**Comment by vividos** [comment]
Thanks for the hint. I updated my port, but unfortunately the build run by `vcpkg_install_msbuild` still fails, not finding the boost headers. The path to the vcpkg include folder isn't passed in the msbuild command line.
I also tested what `USE_VCPKG_INTEGRATION` does, and adding this worked for finding the Boost includes, but now the port is treated as header-only and no .lib and .pdb files are installed.

**Comment by vividos** [comment, marked as answer]
Ok I tried some more changes, and I think `USE_VCPKG_INTEGRATION` is definitely necessary when building a port that's also using `vcpkg.json` in Manifest mode. That my library was considered header-only was a red herring and came from the fact that `vcpkg_install_msbuild` couldn't find the built binaries.
